vent/coordinator/coordinator.py

```diff
@@ -72,127 +62,146 @@ def __init__(self, sim_mode=False):
 
     def stop(self):
-        self.stopping.set()
+        """
+        Stop the coordinator.
+        This does a soft stop (not kill a process)
+        """
+        self.control_module.stop()
 
 
 class CoordinatorRemote(CoordinatorBase):
     def __init__(self, sim_mode=False):
         super().__init__(sim_mode=sim_mode)
-        # TODO: pass max_heartbeat_interval
-        self.process_manager = ProcessManager()
-        self.rpc = IPC()
+        # TODO: according to documentation, pass max_heartbeat_interval?
+        self.process_manager = ProcessManager(sim_mode)
+        self.proxy = get_rpc_client()
+        # TODO: make sure the ipc connection is setup. There should be a clever method
+
+    def get_sensors(self) -> Dict[ValueName, SensorValueNew]:
+        sensor_values = pickle.loads(self.proxy.get_sensors().data)
+        res = {
+            ValueName.PIP: SensorValueNew(ValueName.PIP, sensor_values.pip, sensor_values.timestamp,
+                                          sensor_values.loop_counter),
+            ValueName.PEEP: SensorValueNew(ValueName.PEEP, sensor_values.peep, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.FIO2: SensorValueNew(ValueName.FIO2, sensor_values.fio2, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.TEMP: SensorValueNew(ValueName.TEMP, sensor_values.temp, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.HUMIDITY: SensorValueNew(ValueName.HUMIDITY, sensor_values.humidity, sensor_values.timestamp,
+                                               sensor_values.loop_counter),
+            ValueName.PRESSURE: SensorValueNew(ValueName.PRESSURE, sensor_values.pressure, sensor_values.timestamp,
+                                               sensor_values.loop_counter),
+            ValueName.VTE: SensorValueNew(ValueName.VTE, sensor_values.vte, sensor_values.timestamp,
+                                          sensor_values.loop_counter),
+            ValueName.BREATHS_PER_MINUTE: SensorValueNew(ValueName.BREATHS_PER_MINUTE, sensor_values.breaths_per_minute,
+                                                         sensor_values.timestamp, sensor_values.loop_counter),
+            ValueName.INSPIRATION_TIME_SEC: SensorValueNew(ValueName.INSPIRATION_TIME_SEC,
+                                                           sensor_values.inspiration_time_sec, sensor_values.timestamp,
+                                                           sensor_values.loop_counter),
+        }
+        return res
+
+    def get_active_alarms(self) -> Dict[str, Alarm]:
+        return pickle.loads(self.proxy.get_active_alarms())
+
+    def get_logged_alarms(self) -> List[Alarm]:
+        return pickle.loads(self.proxy.get_logged_alarms())
+
+    def clear_logged_alarms(self):
```

Comment:
This should be get_and_clear_alarms(). We want an atomic get_and_clear operation, otherwise if we "get" and then "clear" independently, an alarm might have arrived in between those two operations that will be lost.

Reply:
Yes, I agree. But I feel even get_and_clear_alarms() is not a satisfied API. As we need to provide a way to clear each alarm. Maybe not change this in this PR, and discuss and finalize the functions in our next meeting.